web/packages/core/src/ruffle-player.ts

```diff
@@ -809,6 +809,49 @@ export class RufflePlayer extends HTMLElement {
             console.error("SWF download failed");
         }
     }
+    private openVirtualKeyboard(): void {
+        const initialValue = "Initial";
+        const initialValueLength = initialValue.length;
+        const existingInput = this.shadow.getElementById(
+            "ruffle-virtual-keyboard"
+        );
+        if (!existingInput) {
+            const input = document.createElement("input");
+            input.setAttribute("type", "text");
+            input.setAttribute("autocapitalize", "off");
+            input.style.position = "absolute";
+            input.style.top = "-100px";
+            input.style.width = "1px";
+            input.style.height = "1px";
+            input.id = "ruffle-virtual-keyboard";
+            input.value = initialValue;
+            input.addEventListener("input", () => {
+                const isBackspace = input.value.length <= initialValueLength;
+                const string = isBackspace
+                    ? "-"
+                    : input.value.slice(initialValueLength);
```

Comment:
Feels a bit hacky to use "-" for the backspace.

How about:
```
const isBackspace = input.value.length <= initialValueLength;
const chars = isBackspace ? ["Backspace"] : [...input.value.slice(initialValueLength];
for (const char of chars) {
   for (const eventType of ["keydown", "keyup"]) {
    // ...
```

Reply:
Fair enough, I wasn't using "-" until I switched to loop over the characters so multi-character strings worked correctly, and after that without a change a backspace would have type "B" -> "a" -> "c".... This is a good idea instead, just not one I thought of myself.